src/libs/API.js

```diff
@@ -169,13 +92,19 @@ Network.registerResponseHandler((queuedRequest, response) => {
         // of the new response created by handleExpiredAuthToken.
         const shouldRetry = lodashGet(queuedRequest, 'data.shouldRetry');
         if (!shouldRetry || unableToReauthenticate) {
-            queuedRequest.resolve(response);
+            if (queuedRequest.resolve) {
```

Comment:
Just curious, when the `queuedRequest.resolve` is not defined?

Reply:
Sure. I can leave a temporary comment here, but basically...

This callback is getting used to handle responses for both 

- persisted requests (which have no resolve/reject methods)
- requests triggered via the main queue (which should always have promise methods attached)

If it's a persisted request then it has no async completion handler and we cannot call.

Reply:
Oh I see, thanks! Yeah, I think a comment could make it clearer :)

Reply:
Agree that a comment would be valuable here. It's a bit confusing for sure